### ISSUE TITLE
This repo and associated webpage should not exist

### ISSUE THREAD

**Issue by ahnolds** [zint/zint#84]
@trevor @g3rrk @ismaell Strongly suggest you take down the https://zint.github.io/ domain and this repo in favor of [the official website](https://zint.org.uk/ ) and the official source code repo on [Sourceforge](https://sourceforge.net/projects/zint/). Since this repo is woefully out of date, it's not helpful to have it around (and as evidenced by the many issues filed here, people don't realize that this is not the up to date official version. Additionally, the zint.github.io page appears higher up the google search results than the official homepage, adding to the confusion.

**Comment by ismaell** [comment]
@ahnolds "the official"? why do you think this version should be taken down? maybe the sourceforge's fork needs to rename itself, given it it's the second one and it changed the license...

**Comment by ismaell** [comment]
@g3rrk I see you're active on the sourceforge version, did anything change from last time? I think I missed something...

**Comment by ahnolds** [comment]
The README in this very repo says
```
The home of libzint is:
	<http://www.sourceforge.net/projects/zint>
and the home for Zint documentation is:
	<http://www.zint.org.uk>
```

It also happens that the code here hasn't been updated since 2013, whereas the code on the sourceforge project was last updated 3 hours ago. Seems like a pretty compelling case that this repo isn't the place anyone should be going to get Zint...

**Comment by ismaell** [comment]
Yes, the location wasn't updated, because it was unclear what would happen at the time. If I add GPL code and they don't integrate it, then there's justification for having two versions; certainly my changes aren't under a permissive license. I need to look into it in more detail.

**Comment by ismaell** [comment]
Also note the git histories are different because I rewrote it to fix the author names, for attribution purposes, so both certainly diverged at least on that.

**Comment by ismaell** [comment]
Yeah, so the SF repo doesn't contain any of my changes. I'm closing this issue unless @g3rrk has any comment... given that @g3rrk is contributing to the other version I might rename this one later...

**Comment by ahnolds** [comment]
It looks to me like their code is also under GPL. I'd politely suggest that you submit a merge request of your code from e724c24 (looks to me like that's the point where your code ended up in this repo but not on SF) to the SF repo. Again, since you haven't touched this code since 2013, in which time the SF version has about 900 new commits. I'd argue that users are more likely to get value from those 900 than from your ~70 here 8 years ago.

And please don't read this as any kind of attack against you, I just found out the hard way that this repo hadn't been touched in a long time when I got all sorts of compiler errors when I tried to build this (errors that have been patched in the SF code).

**Comment by ismaell** [comment]
No, it's a mixture of GPL and BSD, with all new code being BSD. That said you're absolutely right that this repo needs some updating.

**Comment by mrculler** [comment]
@ahnolds I just had the same thing happen - fixed the compiler errors and got all ready to fork/PR when I realized this repo is ancient and non-official.  Kind of a bummer - being "zint/zint" with a proper readme made it seem official.

**Comment by ismaell** [comment]
On 15/Nov/2021 16:41, mrculler wrote:
> @ahnolds I just had the same thing happen - fixed the compiler errors and got all ready to fork/PR when I realized this repo is ancient and non-official.  Kind of a bummer - being "zint/zint" with a proper readme made it seem official.

That's because this was official. Also, please note the other project is under a different license.

**Comment by ahnolds** [comment]
I'm honestly a bit confused about how this was ever official. The [readme from the very first commit in this repo](https://github.com/zint/zint/blob/f2c9c9a5e6689a7538b1e2ae12d31cf2bd436c1a/readme) references the sourceforge page and zint.org.uk, but nowhere do I see github or this repository mentioned.

As for the license, it's definitely true that this is under GPL and the backend for the sourceforge version is BSD. If your argument for keeping this is to keep a GPL version because you believe strongly in copyleft, totally valid, but I would suggest that this repo (and maybe the group too) be renamed to zint-gpl, and information be added to the readme clarifying that this is the GPL version, and that an alternate (BSD licensed and more recently updated) version is available from the sourceforge/zint.org.uk sites. If that's something you'd be open to, I'm happy to write up a sample readme change and submit a PR with it!

**Comment by ismaell** [comment]
Because it was @g3rrk himself who suggested moving to Github from SF, IIRC... anyway, he always had ownership on the github org.

It's true that this version has been neglected, yet by no means the confusion originates in this side.

My take is if a company shows up and wants to fork a project, have it licensed in a more permissive way, and integrate it into their proprietary solutions, perhaps they should be the ones renaming their version... just a thought.

Anyway, I would like to have a chance to see what's going on with the other version before making any changes, I have no idea what happened since then, and it isn't in the top of my list.

**Comment by indig0F10w** [comment]
Here's a perspective from a first time user trying to find a QR code generator. Zint emerges from search, GitHub link included. Came here, saw no updates for like 5-6 years, no releases... At this point I was about to click close tab and go find another generator but I had to check out the issues. Behold, this is not the repo, this is not updated and it is linked everywhere online as Zint repo. And for the cherry top the URL zint.github.com is a 404 page.

Confusion originates on these sides: github.com/zint/zint and zint.github.com

IMO, delete this GitHub page, it is obsolete, useless, has no releases, just issues which no one is fixing. Meanwhile the updated version is on SourceForge. Whoever is in charge of this page should at minimum put a notice on homepage that this is not maintained and include a link to SourceForge.

**Comment by oehhar** [comment]
Thank you for the proposal.

As I did this:
https://github.com/zint/zint/wiki

I probably have something like write access here.

I am a total git novice and will not touch any command line or files.
So, if there are any hints I may do, please advice.

Thank you and best regards,
Harald

**Comment by gitlost** [comment]
Please remove this fork.

**Comment by woo-j** [comment]
Hi @ismaell

I am, or was, @g3rrk . Unfortunately I no longer have access to that account or the email address associated with it, but I'm happy to provide proof of my identity via private email. I was contacted by @gitlost who made me aware of this thread and the confusion that this version of Zint is causing, and so I guess the time has come to make amends. I'm hopeful that together we can finally get this sorted out.

Firstly I want to sincerely apologise - this whole mess is my fault alone and due simply to the fact I didn't properly think through the consequences of what I was doing. As you mention above this was indeed intended to be the "official" version of Zint - and I wanted to move it from SF to GitHub, but I never figured out exactly how to do that. Then I got hacked, locking me out of various accounts and to add to the confusion decided to change the licenses as you mention. In retrospect this was part of a whole series of poor choices, and I was too wrapped up in what I was doing to step back and realise what a mess I was making for both myself and others. This has been a lesson in how NOT to run an open source project!

I am no longer maintaining the project on SourceForge - I have handed control of it to @gitlost who is pushing it forward in ways I can not. For what it's worth I think it would be great to see this repository finally become "the" Zint repository it was always intended to be. GitHub does have advantages over SourceForge, not least that it gets better ranking in search engines. In my ideal circumstance you both (@github and @ismaell) could arrange for the SF code to be transferred here so this would be the new home for the project. I suggest SF repo should be kept but would become a mirror, and the ticketing system and the mailing list on SF would be wound down. I'm happy to mothball my own mirror which I am not good at keeping up to date anyway.

This does raise the issue of the licenses, for which I'm afraid I have no easy solution. I'm afraid that due to an incident where this project was used without my permission in China I lost faith in open source licenses, and so the argument about which license should be valid seems academic to me. My understanding (and certainly my intention) for the licensing on SF was that the GUI and CLI programs be licensed under GPL, but the library be under the more permissive BSD license. I did this because I, as I understand it, a library which uses GPL puts conditions on programs which use the API, but BSD does not and I thought this restriction would hamper the adoption of Zint. For my part I'm happy to (re?)release the code I've written under different licenses if that's desirable. My policy on this has always been the same and can be summed up, unofficially, as "use this program however you want to, just as long as you don't claim that you own it." This doesn't easily translate into legalese.

So in summary, again I apologise for causing this mess - and I hope we can open negotiations for a solution which better serves everyone's needs.

**Comment by woo-j** [comment]
I have just been made an owner again, thank you @trevor

I have also got rid of my old account @g3rrk which was dead anyway.

I don't plan to do anything else just yet, I would like to give the opportunity for anyone who is interested to comment on my ideas for resolving the situation.

**Comment by woo-j** [comment]
This repository is now being merged with the one from SourceForge as proposed in #88
